simulate/bed.go

```diff
@@ -5,39 +5,63 @@ import (
 	"github.com/vertgenlab/gonomics/numbers"
 )
 
+// CountWindows counts the total viable windows of length regionLength in the sequence searchSpace
+func CountWindows(searchSpace []bed.Bed, regionLength int) int {
+	var length, totalWindows int
+	for i := 0; i < len(searchSpace); i++ {
+		length = searchSpace[i].ChromEnd - searchSpace[i].ChromStart
+
+		if length >= regionLength {
+			totalWindows = totalWindows + (length - regionLength + 1)
+		}
+	}
+
+	return totalWindows
+}
+
+// GenerateBedRegion searches the regions of searchSpace (a noGap.bed input file, as a parsed struct) and randomly selects a continuous region of length regionLength 
+func GenerateBedRegion(searchSpace []bed.Bed, tmp int, regionLength int) (bed.Bed, bool) {
+	var chromWindows int
+	var length int
+	// iterating through each ungapped window
+	for j := 0; j < len(searchSpace); j++ {
+		length = searchSpace[j].ChromEnd - searchSpace[j].ChromStart
+
+		// check that the window can fit a sequence of that length e.g. want 10 bp, but window is 5bp
+		chromWindows = length - regionLength + 1
+
+		if chromWindows < 1 {
+			continue
+		}
+		if tmp-chromWindows > 0 {
+			tmp -= chromWindows
+		} else {
+			return bed.Bed{
+				Chrom: searchSpace[j].Chrom, 
+				ChromStart: searchSpace[j].ChromStart + tmp - 1, 
+				ChromEnd: searchSpace[j].ChromStart + tmp - 1 + regionLength, 
+				Name: searchSpace[j].Name,
+				FieldsInitialized: 4}, true
+		}
+	}
+	// TODO: I want it to fatal error if it can't generate answer, but when I had it as a return in the if-else above, it kept on giving "no return" error
```

Comment:
Go functions must end in a return if they are declared as having return variables. In a case like this where you actually want to fatal (or panic, which I would recommend here) you just put the fatal/panic before the return statement. The return is then unreachable, but allows you to compile the function correctly. 

Example:
```
     // stuff above
     log.Panic("error message here")
     return bed.Bed{}, false
}
```

Reply:
Thanks for letting me know, I've added a log.Panic.